mara_db/views.py

```diff
@@ -0,0 +1,196 @@
+        if source_name in shown_tables and target_name in shown_tables:
+            if generate_svg:
+                graph.edge(source_name, target_name)
+            shown_relationships.append((source_name, target_name))
+    if generate_svg:
+        return (graph.pipe('svg').decode('utf-8'), shown_tables, shown_relationships)
+    else:
+        return (None, shown_tables, shown_relationships)
+
+
+@mara_db.route('/svg/<string:db_alias>/<string:schemas>')
+@acl.require_permission(acl_resource)
+def get_svg(db_alias: str, schemas: str):
+    """Shows a chart of the tables and FK relationships in a given database and schema list"""
+    if db_alias not in config.databases():
+        return response.Response(status=400, title=f'unkown database {db_alias}',
+                                 html=f'Error, database {db_alias} is unknown')
+
+    db = config.databases()[db_alias]
+    rendered_svg = draw_schema(db, schemas.split('|'), 'no_columns' in flask.request.args)
+    return flask.jsonify({
+        'svg': rendered_svg[0],
+        'tables': rendered_svg[1],
+        'relationships': rendered_svg[2]
+    })
+
+
+@mara_db.route('/json/<string:db_alias>/<string:schemas>')
+@acl.require_permission(acl_resource)
+def get_structure(db_alias: str, schemas: str):
+    """Retrieve a JSON with the tables and FK relationships in a given database and schema list"""
+    if db_alias not in config.databases():
+        return response.Response(status=400, title=f'unkown database {db_alias}',
+                                 html=f'Error, database {db_alias} is unknown')
+
+    db = config.databases()[db_alias]
+    description = draw_schema(db, schemas.split('|'), 'no_columns' in flask.request.args, generate_svg=False)
+    return flask.jsonify({
+        'tables': description[1],
+        'relationships': description[2]
+    })
+
+
+@mara_db.route('/<string:db_alias>')
+@acl.require_permission(acl_resource)
+def index_page(db_alias: str):
+    """Shows a page to let the user pick some schemas and see the tables and FK"""
+    if db_alias not in config.databases():
+        return response.Response(status=404, title=f'unkown database {db_alias}',
+                                 html=[bootstrap.card(body=_.p(style='color:red')[
+                                     'The database alias ',
+                                     _.strong()[db_alias],
+                                     ' is unknown'
+                                 ])])
+
+    db = config.databases()[db_alias]
+    from mara_db import generic_db_helper
+    available_schemas = generic_db_helper.list_schemas(db)
+    if len(available_schemas) == 0:
+        return response.Response(title=f'No schemas to display for {db_alias}',
+                                 html=[bootstrap.card(
+                                     body=f'The database source {db_alias} has no schemas suitable for displaying (no tables with foreign key constraints)')])
```

Comment:
So if I have no table with a FK contraint, I can't display the schema? Why this restriction?

Reply:
It's the logic present in the old implementation, it hides all tables without FK. Could it be the case to just display every table? I don't know the rationale for the first one. @martin-loetzsch maybe does